### PR TITLE
net aspire volumens

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-dev/issues/3515

### CODE REVIEW

**Review by weikanglim — APPROVED** [review]
LGTM! Really cool that we're now supporting persistent stores with relative few lines of changes.

**Comment by DamianEdwards** [comment]
I tried this with a postgres container but it fails to initialize itself due to permissions errors when it tries to chmod paths in the mounted volume on start:

```
2024-03-21T00:05:25.05723  Connecting to the container 'catalog'...
2024-03-21T00:05:25.07435  Successfully Connected to container: 'catalog' [Revision: 'catalog--rys4cd9-976fb8754-hb5jh', Replica: 'catalog--rys4cd9']
2024-03-21T00:05:21.595201640Z chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
2024-03-21T00:05:21.629449879Z The files belonging to this database system will be owned by user "postgres".
2024-03-21T00:05:21.629499942Z This user must also own the server process.
2024-03-21T00:05:21.629503118Z 
2024-03-21T00:05:21.629597834Z The database cluster will be initialized with locale "en_US.utf8".
2024-03-21T00:05:21.629604998Z The default database encoding has accordingly been set to "UTF8".
2024-03-21T00:05:21.629613804Z The default text search configuration will be set to "english".
2024-03-21T00:05:21.629617281Z 
2024-03-21T00:05:21.629620387Z Data page checksums are disabled.
2024-03-21T00:05:21.629625626Z 
2024-03-21T00:05:21.641722725Z initdb: error: could not change permissions of directory "/var/lib/postgresql/data": Operation not permitted
2024-03-21T00:05:21.642571728Z fixing permissions on existing directory /var/lib/postgresql/data ... 
```

**Comment by vhvb1989** [comment]
@DamianEdwards , that issue looks unrelated to mounting the volume, but to the container itself. See: https://stackoverflow.com/questions/56188573/permission-issue-with-postgresql-in-docker-container

File: cli/azd/resources/apphost/templates/resources.bicept [hunk 1]
```diff
@@ -178,11 +232,22 @@ resource {{bicepName $name}} 'Microsoft.App/containerApps@2023-05-02-preview' =
...
+{{- end}}
       ]
 {{- end}}
     }
     template: {
+{{- if $value.Volumes}}
+      volumes: [
+{{- range $volume := $value.Volumes}}
+        {
+          name: '${toLower('{{$name}}')}-${toLower('{{$volume.Name}}')}'
```
[the remark this answers is not in the document]
Comment (vhvb1989) [reply cli/azd/resources/apphost/templates/resources.bicept:244]: It is just a little easier to handle those problematic names with dashes directly on the template (it is just a decoration for the name in the go template), than adding an extra step when loading/compiling the manifest to memory to handle the naming-fixing.
Fixing the naming as part of loading/compiling means adding logic to detect/find the fields/places where the renaming needs to happen.
And then, there is a function dotnet_importer.Services which only reads the manifest file directly and pull the projects and references w/o compiling (resolving values).  We would need to compile there as well to do the name resolving.
Basica, parsing the manifest only into memory become useless as a source of truth without compiling if we do the renaming on compile step.

**Comment by DamianEdwards** [comment]
Seems this is a known issue when using Azure Files shares and mount volumes in containers: https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/could-not-change-permissions-azure-files

Another post: https://learn.microsoft.com/en-us/answers/questions/89827/how-can-i-change-folder-or-file-permissions-when-m

**Pull request merged**